### ISSUE TITLE
Licensing issues

### ISSUE THREAD

**Issue by blag** [stephenmcd/django-email-extras#29]
Hi, I'm trying to figure out how the GPL plays with Python imports.

I'm working on a closed-source project where we want to sign and encrypt certain email (read: password reset emails, notifications of changes to sensitive settings, suspicious authentication/authorization notifications, etc., but especially password reset emails).

I'd like to use django-email-extras because it seems to be the most mature and maintained project that supports this, and it's licensed under the BSD-3 clause license.

However, it depends on and imports Python code from python-gnupg, which is licensed under the GPLv3. And the GPL being copyleft, that would seem to imply that - as long as this project depends on and imports code from python-gnupg - this project could only be licensed under a GPLv3 compatible license. And the BSD-3 clause is compatible, but if I depend on or import code from django-email-extras, the viral nature of copyleft licenses would mean that my code would then have to be licensed under a GPLv3 compatible license.

Can anybody assuage my fears of using this project? Does the FSF consider Python `import`s to be dissimilar from C's `#include`s and binary linking? Is there something I'm not seeing that prevents the viral nature of the GPL from subsuming code that `import`s code from this project?

Or should I keep looking for a similar project with completely compatible licensing requirements?

Thanks!

**Comment by stephenmcd** [comment]
No idea.

**Comment by blag** [comment]
🙁 

Thanks! I'll keep looking.

**Comment by blag** [comment]
Just in case anybody cares, I think I figured the licensing situation out.

TL;DR: This project does not import any GPLed code, so this issue is moot.

There are two different major forks of python-gnupg:

1. The original fork is [Vinay Sajip's fork](https://bitbucket.org/vinay.sajip/python-gnupg) (old [Google Code repo](https://code.google.com/archive/p/python-gnupg/), [mailing list](https://groups.google.com/forum/#!forum/python-gnupg)). This is also the fork that is registered as [python-gnupg on PyPI](https://pypi.python.org/pypi/python-gnupg). This fork is [3-clause BSD licensed](https://bitbucket.org/vinay.sajip/python-gnupg/src/afbe4e74cfc7fb79dde344499e8934c88874ef85/LICENSE.txt?fileviewer=file-view-default) since at least version 0.2.7 ([Google Code Issue on license clarification](https://code.google.com/archive/p/python-gnupg/issues/24)).
2. The second fork is [Isis Agora Lovecruft's fork](https://github.com/isislovecruft/python-gnupg), "including security patches, extensive documentation, and extra features". When Isis forked from Sajip in version 0.3.2, for some reason the [license](https://github.com/isislovecruft/python-gnupg/commit/69b7f865299db2cda71b9ba0f4319eefec908f9d#diff-9879d6db96fd29134fc802214163b95a) was AGPLv3. The license was later changed to GPLv3 in [commit 6b154c294e373f167de600378e1abd00ad8b0a83](https://github.com/isislovecruft/python-gnupg/commit/6b154c294e373f167de600378e1abd00ad8b0a83#diff-9879d6db96fd29134fc802214163b95a) ([full commit history of the LICENSE file](https://github.com/isislovecruft/python-gnupg/commits/master/LICENSE)).

I've dug through the 0.2.6 and 0.3.2 tarballs from the [Google Code downloads page](https://code.google.com/archive/p/python-gnupg/downloads), and both are licensed under the 3-clause BSD.

Luckily, django-email-extras depends on the version in PyPI (Sajip's version), which is 3-clause BSD licensed, so we can entirely ignore the interplay between Python modules and the GPL.
